src/ampycloud/data.py

```diff
@@ -354,36 +391,47 @@ def metarize(self, which: int = 'slices', base_lvl_alt_perc: float = 5.,
         # Prepare a pandas DataFrame to store all the info
         pdf = pd.DataFrame(index=range(n_ind), columns=cols)
 
-        # What are the original sli/gro/lay ids ?
-        oids = np.unique(self.data[which[:-1] + '_id'])
-
-        # For the moment, happily ignore anything that was not assigned to a sli-gro-lay
-        # WARNING: *if* the chosen clustering approach changes, one may need to start keeping track
-        # of hits that do not get assigned to a sli/gro/lay.
-        oids = np.delete(oids, np.where(oids == -1))
-
-        # Then loop through all of them and extract the required info for each
-        for (ind, oid) in enumerate(oids):
-
-            # Keep track of the original sli/gro/lay id
-            pdf.loc[ind, 'original_id'] = oid
+        original_ids = self._get_original_ids(which)
 
+        for ind, oid in enumerate(original_ids):
             if which == 'groups':
                 # Here, check if the layering was already done ... in which case one should NOT
                 # be metarizing clusters ! This is one of those places where it is assumed that
                 # the layering step comes *after* the grouping step.
                 if self._layers is not None:
-                    raise AmpycloudError('Layering already done.' +
-                                         ' If you metarize your groups now, you will loose the' +
-                                         ' layering information !')
-
+                    raise AmpycloudError(
+                        'Layering already done.'
+                        ' If you metarize your groups now, you will loose the'
+                        ' layering information !'
+                    )
                 # If all is as expected, then set the number of sub-components to -1 for now, until
                 # the layering step decides otherwise (possibly).
                 pdf.loc[ind, 'ncomp'] = -1
+            # Keep track of the original sli/gro/lay id
+            pdf.loc[ind, 'original_id'] = oid
+
+        return pdf, original_ids
+
+    def _calculate_cloud_cover(
+            self, which: str, pdf: pd.DataFrame, original_ids: npt.ArrayLike
+        ) -> pd.DataFrame:
+        """Calculate cloud cover for a given slice, group or layer.
+
+        Args:
+            which (str): One of 'slices', 'groups' or 'layers'
+            pdf (pd.DataFrame): A data frame with slices/ groups/ layers.
+            original_ids (npt.ArrayLike): the original IDs of the slices/ groups/
+                layers.
 
+        Returns:
+            pd.DataFrame: The input data frame with results in the okta column.
+
+        Results are written to the "okta" column of the DF.
+
+        """
+        for ind, oid in enumerate(original_ids):
```

Comment:
Can this function be used also to recalculate the amount from the merged groups/layers? (i.e. not the original_ids)

Reply:
not sure I get the comment. This function is used in the `metarize()` public method to calculate the cloud amount and `metarize()` is called after the merging steps.

Reply:
maybe rename `original_id` to `cluster_id`

Reply:
I would support a renaming. The term "original ID" is definitely not very explicit. ``cluster_id`` sounds good.

Reply:
renamed to `cluster_id`